src/libmongoc/src/mongoc/mongoc-openssl.c

```diff
@@ -702,12 +703,39 @@ _contact_ocsp_responder (OCSP_CERTID *id, X509 *peer)
          GOTO (retry);
       }
 
-      if (!(resp = OCSP_sendreq_bio (bio, path, req))) {
-         MONGOC_DEBUG (
-            "Could not perform an OCSP request for url '%s'. Error: %s",
-            url,
-            ERR_STR);
+      /* Leave OCSP request NULL, set it onto the request context after setting
+       * the host header. */
+      sendreq_ctx =
+         OCSP_sendreq_new (bio, path, NULL /* OCSP request */, 0 /* maxline */);
```

Comment:
This now creates an `OCSP_REQ_CTX` to wrap around the `OCSP_REQUEST` and add the `Host` HTTP header.

Reply:
Why not just move this line of code under the `host` check (to line 718) and _then_ add the request? That way we could eliminate lines 718-721.

Reply:
`man OCSP_REQ_CTX_set1_req` says:

```
OCSP_REQ_CTX_set1_req() sets the OCSP request in rctx to req. This function should be called after any calls to OCSP_REQ_CTX_add1_header().
```

I interpreted that to mean headers must be set before setting the request, even if the request is set by `OCSP_sendreq_new`. Though it isn't totally clear from the man pages, this seemed safer.

Reply:
Got it.